tests/e2e-evm/test/connect.test.ts

```diff
@@ -0,0 +1,50 @@
+import hre from "hardhat";
+import { expect } from "chai";
+import { whaleAddress, userAddress } from "./addresses";
+
+describe("Viem Setup", function () {
+  it("getChainID() matches the configured network", async function () {
+    const publicClient = await hre.viem.getPublicClient();
+
+    const expectedChainId = (() => {
+      switch (hre.network.name) {
+        case "hardhat":
+          return 31337;
+        case "kvtool":
+          return 8888;
+      }
+    })();
+
+    expect(hre.network.config.chainId).to.eq(expectedChainId);
+    expect(await publicClient.getChainId()).to.eq(expectedChainId);
+  });
+
+  it("is configured with whale and user accounts", async function () {
```

Comment:
Clarification question:
Do we set up this basic CI to make this a way to test future changes related to EVM and things like precompiles, etc? So this one should support king of contracts?

Reply:
It's set up to test that all networks have the same configuration and tests that viem is configured correctly for each.  Tho I'm not sure what you mean by kind of contracts?